hibernate-core/src/main/java/org/hibernate/metamodel/mapping/internal/MappingModelCreationHelper.java

```diff
@@ -217,10 +218,12 @@ public static BasicAttributeMapping buildBasicAttributeMapping(
 			}
 			fetchTiming = FetchTiming.IMMEDIATE;
 			fetchStyle = FetchStyle.JOIN;
+			partitioned = value.isPartitionKey() && !( (EmbeddableMappingType) declaringType ).getEmbeddedValueMapping().isVirtual();
```

Comment:
I think that checking if the declaring type is virtual might be too broad. Is the problem that the owned to-one associations are "partitioned" because the join columns refer to partitioned columns or that the unowned to-one association is also "partitioned" because the target column is partitioned?
Either way, we should IMO try to find a different sort of check for this scenario, because even a single join column could be problematic in the same way.
Maybe we have to adapt `ToOneAttributeMapping.hasPartitionedSelectionMapping()` to always return `false` instead?

Reply:
The problem really was that we added additional `partitioned` basic attribute mappings to the owning side of the association when constructing the `VirtualEmbeddedAttributeMapping` synthetic property for the multi-join-column FK. Since this is just a "fake" component reference, I thought this was fine - are there any other cases where the declaring embedded value mapping is virtual and its attributes are actual "real" properties that should be added to the partition key?

`ToOneAttributeMapping.hasPartitionedSelectionMapping()` is not checked anywhere wrt this error (e.g. `UpdateCoordinatorStandard#applyPartitionKeyRestriction` doesn't call that method).